include/LeMonADE/updater/moves/MoveAddScMonomer.h

```diff
@@ -96,7 +73,7 @@ template <class IngredientsType>
 void MoveAddScMonomer::init(const IngredientsType& ing)
 {
     this->resetProbability();
-    particleIndex=ing.getMolecules().size();
+    this->setParticleIndex(ing.getMolecules().size());
```

Comment:
Even the apply function resets the monomer index to size()-1. For instance Feature Attributes uses this
`ingredients.modifyMolecules()[move.getMonomerIndex()].setAttributeTag(move.getType());`
but this is only called after the addmonomer(x,y,z) statement, where index is reset. I guess we first set particle index to size to have a well defined number making some sense, but it is only used in the tests.